File: Laravel/app/Biab/Resources/Checkout.php
<?php

namespace App\Biab\Resources;

use App\Biab\Client;

/**
 * Checkout hand-off. `start()` returns a Stripe Checkout URL — redirect to
 * it with a 303 so the browser re-issues as GET.
 *
 * Payment never touches this app: no card data crosses this process, which
 * is what keeps a Laravel consumer out of PCI scope.
 */
class Checkout
{
    public function __construct(
        private readonly Client $client,
        private readonly ?string $visitorToken = null,
        private readonly ?string $sessionToken = null,
    ) {
    }

    /**
     * @param  array<string, mixed>  $input  successUrl / cancelUrl / customerEmail …
     * @return array<string, mixed>
     */
    public function start(array $input): array
    {
        return $this->client->post('checkout/start', $input, $this->headers());
    }

    /** @return array<string, mixed> */
    public function get(string $sessionId): array
    {
        return $this->client->get('checkout/'.rawurlencode($sessionId), [], $this->headers());
    }

    /** @return array<string, string> */
    private function headers(): array
    {
        $out = [];
        if ($this->visitorToken) {
            $out['X-BIAB-Cart-Visitor'] = $this->visitorToken;
        }
        if ($this->sessionToken) {
            $out['X-BIAB-Session-Token'] = $this->sessionToken;
        }

        return $out;
    }
}

File: Phoenix/lib/bd_starter/bd/resources/checkout.ex
defmodule BdStarter.Bd.Resources.Checkout do
  @moduledoc """
  Checkout hand-off. `start/3` returns a Stripe Checkout URL under
  **`stripeUrl`** — not `url`, which is the obvious wrong guess.

  Redirect with a 303 so the browser re-issues as GET. No card data crosses
  this process, which is what keeps a Phoenix consumer out of PCI scope.
  """

  alias BdStarter.Bd.Client

  def start(client, token, urls) do
    Client.post(client, "checkout/start", urls, headers(token))
  end

  def get(client, token, session_id) do
    Client.get(client, "checkout/#{URI.encode(session_id, &URI.char_unreserved?/1)}", [], headers(token))
  end

  defp headers(nil), do: []
  defp headers(token), do: [{"x-bd-cart-visitor", token}]
end
